src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -379,40 +373,47 @@ bool PolicyHandler::ClearUserConsent() {
 }
 
 uint32_t PolicyHandler::GetAppIdForSending() const {
+  LOG4CXX_AUTO_TRACE(logger_);
   POLICY_LIB_CHECK(0);
   const ApplicationSet& accessor =
       application_manager_.applications().GetData();
-  HmiLevelOrderedApplicationList app_list(accessor.begin(), accessor.end());
-
-  LOG4CXX_INFO(logger_, "Apps size: " << app_list.size());
 
-  for (HmiLevelOrderedApplicationList::const_iterator first = app_list.begin();
-       first != app_list.end();
-       ++first) {
-    if ((*first)->IsRegistered()) {
-      const uint32_t app_id = (*first)->app_id();
-      DeviceParams device_params = GetDeviceParams(
-          (*first)->device(),
-          application_manager_.connection_handler().get_session_observer());
+  HMILevelPredicate has_none_level(mobile_api::HMILevel::HMI_NONE);
+  Applications apps_without_none_level;
+  std::copy_if(
+      accessor.begin(),
+      accessor.end(),
+      std::inserter(apps_without_none_level, apps_without_none_level.begin()),
+      std::not1(has_none_level));
 
-      const bool is_device_allowed = (kDeviceAllowed ==
-                                      policy_manager_->GetUserConsentForDevice(
-                                          device_params.device_mac_address));
-      const bool is_single_app = (1 == app_list.size());
+  LOG4CXX_DEBUG(logger_,
+                "Number of apps with different from NONE level: "
+                    << apps_without_none_level.size());
 
-      if (is_device_allowed && is_single_app) {
-        return app_id;
-      }
+  std::random_shuffle(apps_without_none_level.begin(),
+                      apps_without_none_level.end());
 
-      const bool is_app_in_none =
-          ((*first)->hmi_level() == mobile_apis::HMILevel::HMI_NONE);
+  const uint32_t app_id =
+      GetApplicationIdForPolicyUpdate(apps_without_none_level);
 
-      if (is_device_allowed && !is_app_in_none) {
-        return app_id;
-      }
-    }
+  if (app_id) {
+    return app_id;
   }
-  return 0;
+
+  Applications apps_with_none_level;
+  std::copy_if(
+      accessor.begin(),
+      accessor.end(),
+      std::inserter(apps_with_none_level, apps_with_none_level.begin()),
+      has_none_level);
+
+  LOG4CXX_DEBUG(
+      logger_,
+      "Number of apps with NONE level: " << apps_with_none_level.size());
+
+  std::random_shuffle(apps_with_none_level.begin(), apps_with_none_level.end());
```

Comment:
@LuxoftAKutsan 
This is a requirement statement. I don't quite understand what bug do you mean? Apps protected by accessor, so they all will be alive during choosing, if that was your concern.